### ISSUE TITLE
Require product name field in object model

### ISSUE THREAD

**Issue by zuk3975** [PrestaShop/PrestaShop#19441]
#### Describe the bug

Product object model field "name" is not required. It was probably made to allow creating "ghost" product in product creation.

#### Expected behavior

Make name required in object model. 
This would introduce BC break. The main question if its worth it, or custom validation should be added to AddProductHandler (being migrated for product page in 1.7.8) instead to avoid the BC break?

Open for discussion

#### Additional information

* PrestaShop version: 1.7.7 (targeting 1.7.8 product page migration).
* PHP version: N/A

**Comment by zuk3975** [comment]
@matks, @eternoendless, @jolelievre ?

**Comment by matks** [comment]
I agree with this. I think integrity is more important that the small BC break it will create.
But @eternoendless decision is needed

**Comment by eternoendless** [comment]
Is there a specific reason to require the name at the ObjectModel level?

**Comment by zuk3975** [comment]
> Is there a specific reason to require the name at the ObjectModel level?

We can do it in command level to avoid bc break.
But I guess  the main reason was to have a more robust model :thinking:

**Comment by matks** [comment]
Yes, we actually think the "require" is missing to allow ghost products and since we're going to remove ghost products, we would like to fix the model along the way

**Comment by zuk3975** [comment]
Still need approval.

**Comment by PierreRambaud** [comment]
We should be able to introduce BC break about the requirement of the field name.
